# 파이썬 기초 문법3

연습 문제1  
공무원 시험 결과가 국어, 영어, 한국사, 선택 교과1, 선택 교과2의 점수가 리스트 형태로 
제공됩니다. 이때 한 과목이라도 60점 미만이면 과락입니다. 과락이면 False를 과락이 
아니면 True를 출력하도록 코드를 완성해보세요.

[국어, 영어, 한국사, 선택 교과1, 선택 교과2]  
scores = [88, 73, 89, 59, 77]  
?????  
실행 결과:   
False


In [2]:
scores = [88, 73, 89, 59, 77]

In [8]:
print(scores[0] >=60 and scores[1] >= 60 and scores[2] >= 60 and scores[3] >= 60 and scores[4])

False


print(scores[0] >=60 and scores[1] >= 60 and scores[2] >= 60 and scores[3] >= 60 and scores[4])

## 파이썬 입력 사용해보기

In [1]:
input_value = input("message")

message 1111


input_value = input("message")  
message 1111

변수 입력 받아서 할때 format 
보통은 f스트링 쓴다.

In [28]:
num = int(input("정수를 입력하세요"))
num2 = int(input("정수를 입력하세요"))
print("첫번째 정수는 {} 입니다.".format(num))
print("두번째 정수는 {} 입니다.".format(num2))
print(num,"과",num2,"의 합은",num+num2,"입니다.")


정수를 입력하세요300
정수를 입력하세요400
첫번째 정수는 300 입니다.
두번째 정수는 400 입니다.
300 과 400 의 합은 700 입니다.


사용자의 이름을 물어보고 이어서 2개의 정수를 받아서 덧셈을 한 후 결과를 출력하는 프로그램을 작성해보세요.

이름을 입력하시오: 엔코아  
엔코아 씨, 안녕하세요?  
파이썬에 오신 것을 환영합니다.  
첫 번째 정수를 입력하세요: 300  
두 번째 정수를 입력하세요: 400  
300 과 400 의 합은 700 입니다 

해답) name = input("이름을 입력하시오")  
print(name,"씨, 안녕하세요?")  
print("파이썬에 오신 것을 환영합니다.")  
a = int(input("첫 번째 정수를 입력하세요"))  
b = int(input("두 번째 정수를 입력하세요"))  
print(a,"과",b,"의 합은",(a + b), "입니다")  

In [7]:
name = input("이름을 입력하시오")
print(name,"씨, 안녕하세요?")
print("파이썬에 오신 것을 환영합니다.")
a = int(input("첫 번째 정수를 입력하세요"))
b = int(input("두 번째 정수를 입력하세요"))
print(a,"과",b,"의 합은",(a + b), "입니다")

이름을 입력하시오엔코아
엔코아 씨, 안녕하세요?
파이썬에 오신 것을 환영합니다.
첫 번째 정수를 입력하세요300
두 번째 정수를 입력하세요400
300 과 400 의 합은 700 입니다


In [19]:
stadium = input("경기장은 어디입니까?")
winner = input("이긴 팀은 어디입니까?")
loser = input("진 팀은 어디입니까?")
mvp = input("우수 선수는 누구입니까?")
score = input("스코어는 몇대몇입니까?")

print("\n[속보]",winner,"가",score,"로 승리...")
print("오늘",stadium,"경기장에서 야구경기가 있었습니다.")
print(winner,"와",loser,"는 치열한 공방전을 펼쳤습니다.")
print("이 경기에서",mvp,"가 맹활약을 했습니다.")
print("그 끝내",winner,"가",loser,"를",score,"로 승리하였습니다")
print("기사 끝")
     

경기장은 어디입니까?고척
이긴 팀은 어디입니까?kt
진 팀은 어디입니까?lg
우수 선수는 누구입니까?강백호
스코어는 몇대몇입니까?10:9

[속보] kt 가 10:9 로 승리...
오늘 고척 경기장에서 야구경기가 있었습니다.
kt 와 lg 는 치열한 공방전을 펼쳤습니다.
이 경기에서 강백호 가 맹활약을 했습니다.
그 끝내 kt 가 lg 를 10:9 로 승리하였습니다
기사 끝


입력 값을 여러 개의 변수에 저장하기 

예제) split(',')

In [22]:
result = '1,2,3'. split(',')
print(type(result))
print(result)

<class 'list'>
['1', '2', '3']


## 연습문제

사용자로부터 “hh:mm:ss” 형태의 시간 값을 입력 받습니다. 예를 들면 12시 33분 23초일 때 “12:33:23” 과 같이 입력을 받습니다. 이 때 결과가 아래와 같게 나오도록 
구현해보세요.

hh:mm:ss 의 형태로 시간을 입력해주세요 12:33:23  
실행 결과:  
시 : 12  
분 : 33  
초 : 23

In [35]:
h, m, s = input("hh:mm:ss 의 형태의 시간 값을 입력 받습니다.").split(":")

print("시:",h)
print("분:",m)
print("초:",s)

hh:mm:ss 의 형태의 시간 값을 입력 받습니다.12:33:23
시: 12
분: 33
초: 23


## map() 내장 함수 이용하기


연습 문제  
사용자로부터 값을 입력 받습니다. 국어, 영어, 수학, 과학 점수를 한번에 입력받습니다.  이를 kor, eng, mat, sci라는 변수에 각각 저장하고, 사칙 연산을 활용하여 평균 점수를 
산출해보세요.  
국어, 영어, 수학, 과학 점수를 입력해주세요 98,100,96,100  
실행 결과:  
평균 점수는 98.5 점입니다

In [54]:
kor, eng, mat, sci = map(int,input("국어,영어,수학,과학 점수를 입력해주세요").split(","))
avg = (kor+ eng+ mat+ sci)/4
print("평균 점수는" ,avg, "점입니다.")

국어,영어,수학,과학 점수를 입력해주세요98,100,96,100
평균 점수는 98.5 점입니다.


In [56]:
year = '2021'
month = '02'
day = '03'
hour = '18'
minute = '33'
second = '23'

print(year, month, day, end="")
print(hour, minute, second, sep=":")

2021 02 0318:33:23


In [64]:
app, man = map(int,input("사과와 귤을 몇개 드릴까요?").split(","))
s = (app*1000) + (man*200)
print("총 가격은",s,"원 입니다.")


사과와 귤을 몇개 드릴까요?10,20
총 가격은 14000 원 입니다.


if문 예제

In [66]:
input_value4 =4
print(type(input_value4))

<class 'int'>


In [67]:
print(int)

<class 'int'>


In [69]:
type(input_value4) is int

True

In [70]:
if type(input_value4) is int:
    print("input_value4는 정수입니다.")

input_value4는 정수입니다.


In [71]:
age = 39
if 20 <= age < 30:
    print("20대 입니다.")
elif 30 <= age < 40:
    print("30대 입니다.")
    
print("감사합니다.")

30대 입니다.
감사합니다.


연습 문제
사용자로부터 아이디를 입력받습니다. 이때 아이디가 'admin'이면 '최고 관리자, 모든 
권한을 갖습니다.'를 출력하고 프로그램을 종료합니다. 아이디가 'admin'이 아니면 회원 
등급을 정수로 입력받습니다. 만약 회원 레벨이 1~3 사이 값이면 '제한적인 권한을 
갖습니다.'를 출력하고, 그렇지 않으면 '권한이 없습니다.'를 출력하는 프로그램을 
작성해보세요.  
실행 결과:  
아이디를 입력하세요 : biana  
회원 등급을 입력해주세요 : 3  
제한적인 권한을 갖습니다

코딩) name = input('아이디를 입력하세요 :')  

if name == 'admin':  
    print("최고 관리자, 모든 권한을 갖습니다.")  
else:  
    grade = int(input('회원 등급을 입력해주세요:'))  
    
    if 1<= grade <=3:  
        print("제한적인 권한을 갖습니다.")
    
    else:  
        print("권한이 없습니다.")
    

In [76]:
name = input('아이디를 입력하세요 :')

if name == 'admin':
    print("최고 관리자, 모든 권한을 갖습니다.")
else:
    grade = int(input('회원 등급을 입력해주세요:'))
    
    if 1<= grade <=3:
        print("제한적인 권한을 갖습니다.")
    
    else:
        print("권한이 없습니다.")
    

아이디를 입력하세요 :biana
회원 등급을 입력해주세요:3
제한적인 권한을 갖습니다.
